andela/bp-esa-frontend
# 130635935(sub-task) Implement filter with date

#### What does this PR do?
- add a date picker component
- modify the data structure holding the filters
- made the filter component more flexible(render different types of filters)
- create different functions to handle different types of filtering
- modify the `constants.js` holding the configuration of the filters
#### Description of Task to be completed?
- User should be able to filter automation reports with *automation date*
#### How should this be manually tested?
- Clone and set up the project using the instruction in the README.
- Use `$ git checkout ft-filter-report-with-date` to check out the branch containing the changes of this pull request
- Use `$ npm start` to start the application
- Visit `http://localhost:3000`/ to see the report page.
- You should see the filter and also be able to interact with the different filter options
#### What are the relevant pivotal tracker stories?
This task is a sub-task under the story #130635935
#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-09-27 at 10 25 39 am" src="https://user-images.githubusercontent.com/26048536/46136734-32c0ed00-c240-11e8-8a37-5101f61e15fe.png">
<img width="1440" alt="screen shot 2018-09-27 at 10 30 36 am" src="https://user-images.githubusercontent.com/26048536/46137006-baa6f700-c240-11e8-8046-fffcad52188d.png">
